tests/selection.py

```diff
@@ -50,14 +50,42 @@ def test_bad_X(self):
         with self.assertRaises(Exception):
             self.model(X=X[:10], mixing=0.5, Y=Y)
 
+    def test_quick_run(self):
+        """
+        This test checks that the model can select 1 index
+        """
+        X, Y = load_boston(return_X_y=True)
+
+        model = self.model(X=X, mixing=0.5, Y=Y)
+        model.select(n=2)
+        self.assertEqual(len(model.idx), 2)
+
+    def test_negative_select(self):
+        """
+        This test checks that the model errors when asked to select negative indices
+        """
+        X, Y = load_boston(return_X_y=True)
+
+        with self.assertRaises(ValueError):
+            model = self.model(X=X, mixing=0.5, Y=Y)
+            model.select(n=-1)
+
 
 class FeatureFPSTest(unittest.TestCase, PCovSelectionTest):
-    def setup(self):
+    def setUp(self):
         self.model = FeatureFPS
 
+    def test_supplied_indices(self):
+        """
+        This test checks FPS will accept pre-defined indices.
+        """
+        X, Y = load_boston(return_X_y=True)
+        model = self.model(X=X, mixing=0.5, Y=Y, idxs=[0, 1])
+        model.select(n=2)
```

Comment:
Same, it would be nice to check which indexes are selected

Reply:
> Ok, I'm done messing with this branch!

 I added a few more tests, but it's not really realistic to check all the indices for sample selection, given that many samples are effectively equivalent and can be chosen randomly in place of one another. I did add it for FS, for which there are only 13 features.

![image](https://user-images.githubusercontent.com/47536110/100112058-1a27de00-2e6f-11eb-980b-cc2067d1b668.png)